lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1577,13 +1577,19 @@ def cmd_set(*args)
     name  = args[0]
     value = args[1, args.length-1].join(' ')
 
-    # Set PAYLOAD by index
+    # Set PAYLOAD
     if name.upcase == 'PAYLOAD' && active_module && (active_module.exploit? || active_module.evasion?)
-      index_from_list(payload_show_results, value) do |mod|
-        return false unless mod && mod.respond_to?(:first)
+      if value.start_with?(%r{^(/?payload/)|^/}i)
```

Comment:
Can this be simplified to something like [this](https://rubular.com/r/chlwPx6POhj4ci) without alternation? [Original](https://rubular.com/r/RCYVMfCyDM6DNs) for comparison. Non-capturing group is desired if you're not using the match.

Reply:
```ruby
if value.start_with?('/', 'payload/')
  value.sub!(%r{^/?(?:payload/)?}, '')
end
```

Reply:
This won't run the `sub!` unless your preconditions are met by the slightly faster `start_with?` string check.

Reply:
https://medium.com/@Bakku1505/ruby-start-with-end-with-vs-regular-expressions-59728be0859e

Reply:
Apologies, my comment was directed at this entire block of code, not just this line.

Reply:
Implemented 
```ruby
if value.start_with?('/', 'payload/')
  value.sub!(%r{^/?(?:payload/)?}, '')
end
```
for efficiency & readability sake